scripts/silkit_deb.py

```diff
@@ -0,0 +1,186 @@
+        else:
+            logger.info("Found the debian dir!")
+
+    def __create_orig_tarball(self):
+        silkit_version = self.build_info.version
+
+        if silkit_version == None:
+            raise RuntimeError("No valid SilKit Version found! Exiting!")
+
+        tarball_name = "libsilkit_{}.{}.{}.orig.tar.gz".format(
+            silkit_version.major, silkit_version.minor, silkit_version.patch
+        )
+
+        try:
+            subprocess.run(
+                [
+                    "tar",
+                    "--exclude=.git",
+                    "-czf",
+                    self.build_info.work_dir / tarball_name,
+                    "-C",
+                    self.build_info.silkit_info.silkit_source_path,
+                    ".",
+                ],
+                check=True,
+            )
+        except Exception as ex:
+            raise RuntimeError(f"While creating the Source tarball, an error occured!\n{str(ex)}")
+
+    def __copy_debian_dir(self):
+        try:
+            shutil.copytree(
+                self.build_info.silkit_pkg_path.expanduser() / "debian/",
+                self.build_info.work_dir / "sil-kit/debian",
+            )
+        except Exception as ex:
+            logger.error("Could not copy the debian dir into the sil kit source dir! Exiting")
+            raise RuntimeError(f"{str(ex)}")
+
+    def __parse_platform(self) -> str:
+
+        dist, version = self.build_info.platform.split("-")
+        logger.info(f"Building for {dist} version {version}")
+
+        return version
+
+    def __build_package(self, build_flags: BuildFlags):
+
+        try:
+            platform_build_flags = (
+                f"--set-envvar=PLATFORM_BUILD_FLAGS={build_flags.add_platform_flags}"
+                if build_flags.add_platform_flags != ""
+                else ""
+            )
+
+            debuild_flags = build_flags.add_debuild_flags.split(" ")
+            logger.debug(f"Additional debuild flags: {debuild_flags}")
+            debuild_cmd = (
+                ["debuild"]
+                + debuild_flags
+                + [
+                    platform_build_flags,
+                    f"--set-envvar=CC={build_flags.c_compiler}",
+                    f"--set-envvar=CXX={build_flags.cxx_compiler}",
+                    "-us",
+                    "-uc",
+                    "--lintian-opts",
+                    "-E",
+                    "--pedantic",
+                ]
+            )
+
+            # If we have '' empty strings in the command list, strange things happen in
+            # exec land
+            debuild_cmd = list(filter(lambda arg: arg != "", debuild_cmd))
+
+            logger.debug(f"CMD LIST: {debuild_cmd}")
+            logger.debug(f"Calling: {' '.join(debuild_cmd)}")
+            subprocess.run(debuild_cmd, check=True, cwd=self.build_info.work_dir / "sil-kit/")
+        except Exception as e:
+            raise RuntimeError(str(e))
```

Comment:
```suggestion
            raise RuntimeError("failed to build package") from e
```

Which adds the exception `e` to the exception context directly. See [the Python docs][1] under _Exception context_.

[1]: https://docs.python.org/3/library/exceptions.html#built-in-exceptions

Reply:
Similarly for the other `raise` statements 😸